code/modules/clothing/rogueclothes/mask/mask.dm

```diff
@@ -40,6 +40,20 @@
 		take_damage(11, BRUTE, "blunt", 1)
 	..()
 
+/obj/item/clothing/mask/rogue/spectacles/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(!user.has_flaw(/datum/charflaw/badsight))
+		if(slot == SLOT_WEAR_MASK && user.STAPER < 5)
+			user.STAPER += 4
+			cured_eyesight = TRUE
+	user.update_fov_angles()
+
+/obj/item/clothing/mask/rogue/spectacles/dropped(mob/living/carbon/human/user)
+	. = ..()
+	if(cured_eyesight)
+		user.STAPER -= 4
+		cured_eyesight = FALSE
+	user.update_fov_angles()
```

Comment:
```suggestion
			user.change_stat("perception", 0, "spectacles")
```
this just sets the spectacles bonus to 0 (undoes the existing bonus), and will call `update_fov_angles()`. it is totally safe if the bonus has not been given, and means you don't need the `cured_eyesight` var at all